Strata/Languages/Boogie/Examples/HeapReasoning.boogie.st

```diff
@@ -0,0 +1,254 @@
+program Boogie;
+// lake exe StrataVerify Strata/Languages/Boogie/Examples/HeapReasoning.boogie.st 
```

Comment:
Something I'd really like to do is have the CI run commands like this, in addition to what it already does, for all standalone Strata source files that we have in the repo.

It makes me wonder whether this file should go in the top-level `Examples` directory, too. Then it's easy to just run on all files in that directory (and that way everything under `Strata` is a Lean source file).